### ISSUE TITLE
Python exception when running some Rust tests via `meson test`

### ISSUE THREAD

**Issue by vanc** [mesonbuild/meson#13967]
I converted a Rust crate to meson and when running tests, Python exception ocurred.

It's version 1.5.1 and the critical information is here.
```
mesonbuild/mtest.py", line 1144, in parse
    _, name, _, result = line.rstrip().split(' ')
    ^^^^^^^^^^^^^^^^^^
ValueError: too many values to unpack (expected 4)
```

Based on the Cargo test, it's clear that the result parsing is too restrictive. It expected exactly 4 fields, but could be more. For example:

```
test digest::tests::max_input::SHA224::too_long_input_test_block - should panic ... ok
```

Rust will add "- should panic" to the output if a test is marked with "should_panic". For example:

```rust
    #[test]
    #[should_panic(expected = "digest update failed")]
    fn too_long_input_test_block() {
        super::too_long_input_test_block(&digest::$algorithm_name);
    }
```

I made a quick fix and it works fine for me.

```patch
diff --git a/mesonbuild/mtest.py b/mesonbuild/mtest.py
index c417bc0b3..aef2534ab 100644
--- a/mesonbuild/mtest.py
+++ b/mesonbuild/mtest.py
@@ -1146,7 +1146,8 @@ class TestRunRust(TestRun):
         n = 1
         async for line in lines:
             if line.startswith('test ') and not line.startswith('test result'):
-                _, name, _, result = line.rstrip().split(' ')
+                lst = line.rstrip().split(' ')
+                name, result = lst[1], lst[-1]
                 name = name.replace('::', '.')
                 t = parse_res(n, name, result)
                 self.results.append(t)
```

**Comment by vanc** [comment]
Fixed by commit 2fd0dac. Closing.

**Comment by eli-schwartz** [comment]
Oh hmm neat, nice side effect. :D
